colossalai/tensor/_ops/addmm.py

```diff
@@ -1,64 +1,60 @@
 import torch
-from typing import Union
 from colossalai.tensor.op_wrapper import colo_op_impl
 from colossalai.nn.layer.parallel_1d._utils import reduce_input, reduce_grad
 from colossalai.tensor import ComputePattern, TensorSpec, ComputePattern, ParallelAction, ColoTensor
 from colossalai.tensor import dist_spec
+from ._utils import GeneralTensor, Number, convert_to_colo_tensor
 
 
-def colo_addmm_1Drow(input_tensor: ColoTensor, mat1: ColoTensor, mat2: ColoTensor, beta: Union[int, float],
-                     alpha: Union[int, float]) -> ColoTensor:
+def colo_addmm_1Drow(input_tensor: ColoTensor, mat1: ColoTensor, mat2: ColoTensor, beta: Number,
+                     alpha: Number) -> ColoTensor:
     parallel_action = mat2.spec.get_action_by_compute_pattern(ComputePattern.TP1D)
     # mat1:S[1] x mat2:S[0] = Output:P
     # beta * input + alpha * All-Reduce(Output) = res
 
-    mat1.to_dist_spec(dist_spec.shard(mat2.spec.get_process_group(), [-1], [mat2.spec.get_process_group().size()]))
+    mat1 = mat1.convert_to_dist_spec(
```

Comment:
sanity check for args, they are in type as you expected.

Reply:
`torch.addmm` and `ColoTensor.__torch_function__` will check types.